### PR TITLE
C++: Reduce duplication from cpp/path-injection

### PR DESCRIPTION
**Pull request by geoffw0**
Reduce duplication from the `cpp/path-injection` query, by making sources and sinks barriers.

### CODE REVIEW

File: cpp/ql/src/Security/CWE/CWE-022/TaintedPath.ql [hunk 1]
```diff
@@ -88,6 +88,11 @@ module TaintedPathConfig implements DataFlow::ConfigSig {
       hasUpperBoundsCheck(checkedVar)
     )
   }
+
+  predicate isBarrierOut(DataFlow::Node node) {
+    // make sinks barriers so that we only report the closest instance
+    isSink(node)
```
Comment (MathiasVP) [inline cpp/ql/src/Security/CWE/CWE-022/TaintedPath.ql:94]: This only makes the sink a barrier (out), but the PR description intended you also wanted to make the source a barrier (in). Is that intentional?

I don't think making `FlowSource` an in barrier will do anything, so I'm quite happy with this as-is, but I just wanted to point out the discrepancy between the PR description and the code 👍

**Review by MathiasVP — APPROVED** [review]
LGTM once we have a happy DCA run (as happy as DCA can currently be)

File: cpp/ql/src/Security/CWE/CWE-022/TaintedPath.ql (hunk 1 above)
Comment (geoffw0) [reply cpp/ql/src/Security/CWE/CWE-022/TaintedPath.ql:94]: Ah, well spotted - I tried both, it didn't make any difference in practice so I dropped the sources bit.  Looks like I forgot that when I wrote the PR description though!

**Comment by geoffw0** [comment]
DCA shows 61 lost results for `cpp/path-injection`, which is expected, and a very small sample (3) of those results LGTM.  There are a whole lot of failures though, so I'm going to re-run it.

**Comment by MathiasVP** [comment]
> There are a whole lot of failures though, so I'm going to re-run it.

Yeah, there was a DCA issue last week caused by some analysis changes. That should be fixed now. I expect that you'll get many more removed results on this run when vim successfully completes since vim is known to have a lot of results on this query.

**Comment by geoffw0** [comment]
Yep, we're now up to 80 lost results, and again a very small sample LGTM.  There are still some failures, but fewer, and nothing else looks amiss.  Merging.

**Comment by MathiasVP** [comment]
> Yep, we're now up to 80 lost results, and again a very small sample LGTM. There are still some failures, but fewer, and nothing else looks amiss. Merging.

Indeed. These are known Windows tracer issues that Dave is currently looking into

**Pull request merged**